docs/workflows/api-token.md

```diff
@@ -397,3 +431,35 @@ func main() {
 
 </TabItem>
 </Tabs>
+
+## Validating Access Tokens:
+
+There are two types of tokens that can be validated: `self-contained` or `referential`. When validating self-contained tokens, consider if revoked tokens are accepted. If revoked tokens are accepted, then the token can be validated [offline](api-token#offline-validation) by validating the signature and parsing the claims of the JWT token. In all other cases the token should be [introspected](api-token#introspection-online-validation).
+
+### Introspection (Online Validation)
+
+A successful introspect query will return information about the token. This endpoint is compliant with [RFC-7662](https://www.rfc-editor.org/rfc/rfc7662). After recieving information about the token, validate the following:
+
+1. Check that either the application id or resource server id is listed in the `aud` of the token. If at least one of the allowed audiences is in the token `aud` claim, then the token is granted.
+2. Check that token has the expected scopes.
+
+```bash title="/introspect"
+curl https://api-$REGION.beyondidentity.com/v1/tenants/$TENANT_ID/realms/$REALM_ID/introspect \
+-X POST \
+-H "Authorization: Bearer $TOKEN" \
+-H "Content-Type: application/x-www-form-urlencoded" \
+-H 'x-security-context: Security Context containing base64 encoded actor information.' \
+-H 'x-correlation-id: If supplied with the request, the response must contain the same value. If not supplied with the request, it is generated by the server and returned in the response.' \
+-d '{"token":"$TOKEN_TO_INTROSPECT"}'
+```
+
+### Offline Validation
+
+In order to validate a token offline, the JWT header and claims must be decoded. Decode both and follow the steps below:
+
+1. Check that the `jku` host in the JWT header is trusted. It should equal `auth-$REGION.beyondidentity.com`.
```

Comment:
Looking back at this explanation, I think we need to clarify it a little further. Let me know if this sounds clear to you:

> 1. Parse the [host](https://datatracker.ietf.org/doc/html/rfc3986#section-3.2.2) of the URI in the `jku` header field. If it equals `auth-$REGION.beyondidentity.com`, then proceed. Otherwise, reject the token.